### PR TITLE
Fix various map control bugs

### PR DESCRIPTION
**Pull request by Pessimistress**
- Smooth rotation
- Smooth pan
- Zoom around mouse position

### CODE REVIEW

File: src/components/map-controls.js [hunk 1]
```diff
@@ -240,7 +240,7 @@ export default class MapControls extends PureComponent {
   }
 
   _unproject(pos) {
-    const viewport = new PerspectiveMercatorViewport(Object.assign({}, this.props));
+    const viewport = new PerspectiveMercatorViewport(this.props);
     const lngLat = this.props.unproject ?
```
Comment (ibgreen) [inline src/components/map-controls.js]: I think we can remove this.props.unproject now (it was there to allow injecting mapbox version during development)

File: src/components/map-controls.js [hunk 2]
```diff
@@ -251,10 +251,7 @@ export default class MapControls extends PureComponent {
   // TODO - We should have a mapbox-independent implementation of panning
   // Panning calculation is currently done using an undocumented mapbox function
   _calculateNewLngLat({startDragLngLat, pos, startPos}) {
-    const viewport = new PerspectiveMercatorViewport(Object.assign({}, this.props, {
-      longitude: startDragLngLat[0],
-      latitude: startDragLngLat[1]
-    }));
+    const viewport = new PerspectiveMercatorViewport(Object.assign({}, this.props));
```
Comment (ibgreen) [inline src/components/map-controls.js]: Remove the `Object.assign` here as well if we're doing it above?

File: src/components/map-controls.js [hunk 3]
```diff
@@ -282,8 +279,8 @@ export default class MapControls extends PureComponent {
     if (yDelta > 0) {
       // Dragging downwards, gradually decrease pitch
       if (Math.abs(this.props.height - startPos[1]) > PITCH_MOUSE_THRESHOLD) {
-        const scale = yDelta / (this.props.height - startPos[1]);
-        pitch = (1 - scale) * PITCH_ACCEL * startPitch;
+        const scale = yDelta / (this.props.height - startPos[1]) * PITCH_ACCEL;
```
Comment (ibgreen) [inline src/components/map-controls.js]: While I don't doubt this behaves better, this still looks weird to me, is `scale` still the right name after this change (if it ever was) or is there something more intuitive.

Comment (abmai) [inline src/components/map-controls.js]: Perhaps just modify the equation on line 286 instead? i.e. `(1 - scale * PITCH_ACCEL) * startPitch`

**Pull request merged**